### ISSUE TITLE
ir_reciprocal_mesh output different result to QE and VASP

### ISSUE THREAD

**Issue by unkcpz** [atztogo/spglib#79]
I have a cell:

```
(array([[0. , 5.4, 5.4],
        [5.4, 0. , 5.4],
        [5.4, 5.4, 0. ]]), [[0, 0, 0], [0.25, 0.25, 0.25]], [1, 1])
```

When getting ir_reciprocal_mesh with shift, there is only 4 ir-kpoints:

```
mapping, grid = spglib.get_ir_reciprocal_mesh(mesh, cell, is_shift=[1, 1, 1])

Number of ir-kpoints: 4
[[ 0.16666667  0.16666667  0.16666667]
 [ 0.5         0.16666667  0.16666667]
 [ 0.5         0.5         0.16666667]
 [-0.16666667  0.5         0.16666667]]
```

The QE and VASP ouput the same ir-kpoints:

```
Reciprocal lattice
    0.16666666666667    0.16666666666667    0.16666666666667            96
    0.50000000000000    0.16666666666667    0.16666666666667           288
   -0.16666666666667    0.16666666666667    0.16666666666667           288
    0.50000000000000    0.50000000000000    0.16666666666667           288
   -0.16666666666667    0.50000000000000    0.16666666666667           288
    0.50000000000000    0.50000000000000    0.50000000000000            48
```

Is there any problem in the function `spg_get_ir_reciprocal_mesh` ?

**Comment by atztogo** [comment]
[3, 3, 3] grid mesh with half shifts for this cell can not be mapped on it for rotations. The treatment on this type of grid is different in spglib (take less) and VASP/QE (take over). It is recommended to use Gamma centre mesh in this case. But I think it's better for spglib to support the same behaviour as VASP/QE.

**Comment by atztogo** [comment]
I start to doubt what I wrote above. I will consider it carefully.

**Comment by unkcpz** [comment]
Thanks, look forward to your update.

**Comment by atztogo** [comment]
My conclusion is that spglib doesn't consider this case, i.e., the behaviour is undefined. There is no answer that is symmetrically sound than choosing Gamma centred. So the problem is how we should define the behaviour in this case. For VASP or QE, I don't know exactly how they define their behaviours.

**Comment by atztogo** [comment]
Probably I understand the behaviour of VASP and QE, which is not perfect but understandable. I found the same treatment is already implemented in spglib as a special case previously was not applied for this case. To handle this issue, I updated spglib at https://github.com/atztogo/spglib/commit/f002c9dcaba47b004a13f68fc0badf14f4361e22. Now I get the same result with 6 ir-kpoints. This will be released hopefully soon with other fixes.
